src/include/txn.i

```diff
@@ -375,7 +375,12 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
             op->type = WT_TXN_OP_BASIC_COL;
     }
     op->u.op_upd = upd;
-    upd->txnid = session->txn.id;
+
+    /* Use the original txnid for the las inserts */
+    if (S2C(session)->cache->las_fileid != 0 && op->btree->id == S2C(session)->cache->las_fileid)
+        upd->txnid = S2C(session)->cache->las_txnid;
```

Comment:
Given that the transaction code needs to special case inserts into the lookaside file regardless, I'm not convinced we should also use the trick of setting snapshot isolation in the lookaside cursor transaction.

OTOH: I don't see why we need to pull this transaction ID from somewhere special - maybe you can explain that to me face to face?

Reply:
As discussed, the transaction id is required to maintain the information of the original transaction. Removed the snapshot isolation change and kept the cache variable to pass the information.